packages/contracts-bedrock/contracts/test/SafeCall.t.sol

```diff
@@ -39,4 +39,117 @@ contract SafeCall_call_Test is CommonTest {
         assertEq(to.balance, value, "to balance received");
         assertEq(from.balance, 0, "from balance not drained");
     }
+
+    function testFuzz_callWithMinGas_hasEnough_succeeds(
+        address from,
+        address to,
+        uint64 minGas,
+        uint64 value,
+        bytes memory data
+    ) external {
+        vm.assume(from.balance == 0);
+        vm.assume(to.balance == 0);
+        // no precompiles
+        vm.assume(uint160(to) > 10);
+        // don't call the vm
+        vm.assume(to != address(vm));
+        vm.assume(from != address(vm));
+        // don't call the console
+        vm.assume(to != address(0x000000000000000000636F6e736F6c652e6c6f67));
+        // don't call the create2 deployer
+        vm.assume(to != address(0x4e59b44847b379578588920cA78FbF26c0B4956C));
+        // don't send funds to self
+        vm.assume(from != to);
+
+        assertEq(from.balance, 0, "from balance is 0");
+        vm.deal(from, value);
+        assertEq(from.balance, value, "from balance not dealt");
+
+        // Bound minGas to [0, l1_block_gas_limit]
+        minGas = uint64(bound(minGas, 0, 30_000_000));
+
+        vm.expectCallMinGas(to, value, minGas, data);
+        vm.prank(from);
+        bool success = SafeCall.callWithMinGas(to, minGas, value, data);
```

Comment:
I feel like nested calls would be a good way to test this because then eip150 comes into play more

Reply:
Nested calls are tested fairly thoroughly in the invariant tests as well as `test_callWithMinGas_noLeakageLow_succeeds` & `test_callWithMinGas_noLeakageHigh_succeeds`

Reply:
Do nested calls really make a difference? It's always just the same 63/64ths rule for each sub-call.

Reply:
@maurelian You're right, but Foundry tests have a default gas limit of `9223372036854775807`. This means for this test specifically, it will *always* have enough gas to satisfy the min gas limit. In the other tests, we limit the amount of gas in the callframe that invokes `callWithMinGas` to a realistic amount.